### DISCUSSION TITLE
Testing TPI2

### DISCUSSION THREAD

**Discussion by ivanscardaoni1** [dds-jv/foro#238] [Consultas de objetos] [answered]
Buen dia, tengo una consulta respecto a como testear la funcionalidad de lectura de mi adapter del LowLevelFileSystem.
    Dado que el metodo writeSyncFile es void, y no conocemos la implementacion del mismo para saber si genera algun tipo de cambio en el buffer (abstraccion de bytes[] end y start) a diferencia de readSyncFile que lo llena y devuelve la cantidad de leidos. No se me ocurre ni siquiera algo para mockear una funcionalidad, ya que seria asumir cosas de writeSyncFile que no conozco.
    ¿Por donde deberia encarar esos testeos?

**Comment by gastonprieto** [comment, marked as answer]
Buenas @ivanscardaoni1 

> no conocemos la implementacion del mismo para saber si genera algun tipo de cambio en el buffer (abstraccion de bytes[] end y start)

Bien.

Como vos lo que queres saber es si tu funcionalidad esta realizada correctamente, a vos lo único que te interesaría es que se llame a ese mensaje, entonces eso es lo importante a chequear

**Reply by gastonprieto** [reply]
te sirvio @ivanscardaoni1 ?

**Reply by ivanscardaoni1** [reply]
Hola! Si, perdón que no respondí, andaba con mil cosas, pero si, lo comprendí y me ayudo a entender que se testeaba a través de un Verify. Gracias Gaston!
